### DISCUSSION TITLE
Updating from MAUI .NET 7 to .NET 8

### DISCUSSION THREAD

**Discussion by MouratidisA** [dotnet/maui#21100] [General] [answered]
I am currently working on a project in MAUI .NET 7, and I was wondering if it is time to update to .NET 8 to improve various bugs, performance and maybe the overall development experience.

I am using Visual Studio for mac, should I change to Windows pc to use the .NET 8 in Visual Studio, because Visual Studio for mac is scheduled for retirement by August 31, 2024…?

Any advice would be appreciated!

**Comment by nikamvd** [comment]
I migrated my project from .NET 7 to .NET 8; however for me many things has started breaking. UI is all distorted and feels like I'm again back to step where I migrated project from Xamarin.Forms to .NET 7. Any feedback from experts is appreciated.

**Reply by davidortinau** [reply]
@nikamvd please share some specifics about the UI breakage you've experienced.

**Comment by davidortinau** [comment, marked as answer]
@MouratidisA VS Mac works with .NET 8 if you enable the preview feature flag I the preferences, and will continue to work with .NET 8 even after the retirement. I definitely encourage you to use .NET 8.

**Reply by MouratidisA** [reply]
@davidortinau You're right, there is a preview feature for using .NET 8! 
I was wondering if I could work with MAUI .NET 8 from my Mac after the retirement (because sources point that the best approach is to have a Windows pc for the Visual Studio) ?

**Comment by MouratidisA** [comment]
Just followed, [Upgrading .NET MAUI from .NET 7 to .NET 8](https://github.com/dotnet/maui/wiki/Upgrading-.NET-MAUI-from-.NET-7-to-.NET-8) and everything works well!!

Useful note: 
1. Commit and push everything about your project (safety first)
2. Enable the preview feature flag on your VS Mac preferences, _"Use the .NET 8 SDK if installed (requires restart)"_
3. Install the .Net 8 for your Mac
4. run `sudo dotnet workload install maui` on your terminal
5. Delete **bin** and **obj** folders from your project
6. Update your NuGet packages

From my experience, I had to:
- Remove `<uses-sdk android:minSdkVersion="24" android:targetSdkVersion="33" />` from my Android manifest.xml to resolve issues with the minSdkVersion and targetSdkVersion.
- Create a new **attrs.xml** file in "projectName/Platforms/Android/Resources/values" with the following content:

`<resources>
<declare-styleable name='SKCanvasView'> <attr name='ignorePixelScaling' format='boolean'/> </declare-styleable>
</resources>
`
- Reimport missing images and assets in the **.csproj** file.

**Reply by abdullahraiyyan** [reply]
I really appreciate the steps you provided. Upgrading to.NET 8 with.NET MAUI was a breeze because of your guide. As long as I followed your instructions, everything went smoothly. 
By the way, you mentioned creating a new attrs.xml file. Could you elaborate on the necessity of this specific step?  Just trying to figure out what it's for.

**Reply by MouratidisA** [reply]
@abdullahraiyyan the **attrs.xml** file resolves an issue with SKCanvasView :
`Could not find Android Resource '@styleable/SKCanvasView'. Please update to add the missing resource.`

While searching for a solution, I found this [SkiaSharp .Net 8 is supported? ](https://github.com/mono/SkiaSharp/issues/2444#issuecomment-1564090245), and followed the proposed workaround. 

They probably fixed the issue, so after upgrading all your packages, give it a try and remove the file.
